hangman/game.py

```diff
@@ -1,13 +1,116 @@
 from .exceptions import *
-
+import random
 
 class GuessAttempt(object):
-    pass
+    
+    def __init__(self, guess, hit=None, miss=None):
+        self.guess = guess
+        self.hit = hit
+        self.miss = miss
+    
+        if self.hit == True and self.miss == True:
+            raise InvalidGuessAttempt
+    
+    def is_hit(self):
+        if self.hit == None:
+            return False
+        else:
+            return True
+
+    def is_miss(self):
+        if self.miss == None:
+            return False
+        else:
+            return True
 
 
 class GuessWord(object):
-    pass
+    
+    def __init__(self, answer):
+        self.answer = answer
+        self.masked = '*'*len(self.answer)
 
+        if len(self.answer) == 0:
+            raise InvalidWordException
+    
+    def perform_attempt(self, guess):
+        if len(guess) > 1:
+            raise InvalidGuessedLetterException
+
+        guess = guess.lower()
+        self.answer = self.answer.lower()
+        
+        if guess in self.answer:
+            result = ''
+            index = 0
+            for i in self.answer:
+                if i.lower() == guess:
+                    result += guess
+                else:
+                    result += self.masked[index]
+                index += 1
+            self.masked = result
+        
+        if guess in self.answer:
+            attempt = GuessAttempt(guess, hit=True)
+        else:
+            attempt = GuessAttempt(guess, miss=True)
+        return attempt
+    
 
 class HangmanGame(object):
-    pass
+    number_of_guesses = 0
+    WORD_LIST = ['rmotr', 'python', 'awesome']
+    
+    def __init__(self, word_list=WORD_LIST, number_of_guesses=5):
+        self.word_list = word_list
+        self.number_of_guesses = number_of_guesses
+        self.previous_guesses = []
+        self.remaining_misses = number_of_guesses
+        self.word = GuessWord(HangmanGame.select_random_word(word_list))
+    
+    @classmethod
+    def select_random_word(cls, word_list):
+        if word_list == []:
+            raise InvalidListOfWordsException
+
+        cls.word_to_guess = random.choice(word_list)
+        return cls.word_to_guess
```

Comment:
hmmm you do not need to have `cls.word_to_guess` .. this `classmethod` is to return random word so you can simply `return random.choice(word_list)`